executor/point_get.go

```diff
@@ -156,8 +156,15 @@ func (e *PointGetExecutor) get(key kv.Key) (val []byte, err error) {
 		return nil, err
 	}
 	if txn != nil && txn.Valid() && !txn.IsReadOnly() {
-		return txn.Get(key)
+		val, err = txn.GetMemBuffer().Get(key)
+		if err == nil {
+			return val, err
+		}
+		if !kv.IsErrNotFound(err) {
+			return nil, err
+		}
```

Comment:
maybe need more comments to explain this and pessimistic = false still can fall through to snapshot logic?